src/integrations/prefect-kubernetes/prefect_kubernetes/worker.py

```diff
@@ -379,17 +379,34 @@ def prepare_for_flow_run(
             deployment: The deployment associated with the flow run used for
                 preparation.
             flow: The flow associated with the flow run used for preparation.
+            work_pool: The work pool associated with the flow run used for preparation.
+            worker_name: The name of the worker used for preparation.
         """
+        # Save special Kubernetes env vars (like those with valueFrom)
+        special_env_vars = []
+        if isinstance(self.env, list):
+            special_env_vars = [item for item in self.env if "valueFrom" in item]
+            original_env = {}
+            for item in self.env:
+                if "name" in item and "value" in item:
+                    original_env[item["name"]] = item.get("value")
+            self.env = original_env
```

Comment:
the mutation of `self.env` for every flow run makes me uncomfortable - this might be an opportunity for a `gettr` `settr` approach to `self.env` or a special method `self._format_env` that encapsulates the logic?

Reply:
I don't think an actual `setter` pattern would be what we want here since `self.env` is a `Field` but I took a shot at cleaning this up in a way that we may want to stick on `BaseJobConfiguration` eventually since they all have `env`

let me know what you think!

Reply:
i reverted this back to the way I had it originally. happy to revisit the convo around access/mutation of self.env in the future